cpp/src/io/parquet/page_enc.cu

```diff
@@ -1738,16 +1762,29 @@ CUDF_KERNEL void __launch_bounds__(block_size, 8)
               v *= ts_scale;
             }
           }
-          dst[pos + 0] = v;
-          dst[pos + 1] = v >> 8;
-          dst[pos + 2] = v >> 16;
-          dst[pos + 3] = v >> 24;
-          dst[pos + 4] = v >> 32;
-          dst[pos + 5] = v >> 40;
-          dst[pos + 6] = v >> 48;
-          dst[pos + 7] = v >> 56;
+          if constexpr (is_split_stream) {
+            auto const stride     = s->page.num_valid;
+            dst[pos + 0 * stride] = v;
+            dst[pos + 1 * stride] = v >> 8;
+            dst[pos + 2 * stride] = v >> 16;
+            dst[pos + 3 * stride] = v >> 24;
+            dst[pos + 4 * stride] = v >> 32;
+            dst[pos + 5 * stride] = v >> 40;
+            dst[pos + 6 * stride] = v >> 48;
+            dst[pos + 7 * stride] = v >> 56;
+          } else {
+            dst[pos + 0] = v;
+            dst[pos + 1] = v >> 8;
+            dst[pos + 2] = v >> 16;
+            dst[pos + 3] = v >> 24;
+            dst[pos + 4] = v >> 32;
+            dst[pos + 5] = v >> 40;
+            dst[pos + 6] = v >> 48;
+            dst[pos + 7] = v >> 56;
+          }
```

Comment:
unless we're really worried about the extra multiplication, we can avoid the extra repetition here
```suggestion
        auto const stride     = is_split_stream ? s->page.num_valid : 1;
        dst[pos + 0 * stride] = v;
        dst[pos + 1 * stride] = v >> 8;
        dst[pos + 2 * stride] = v >> 16;
        dst[pos + 3 * stride] = v >> 24;
        dst[pos + 4 * stride] = v >> 32;
        dst[pos + 5 * stride] = v >> 40;
        dst[pos + 6 * stride] = v >> 48;
        dst[pos + 7 * stride] = v >> 56;
```

Reply:
I'll benchmark it. Hopefully the compiler is smart enough to figure out it can skip the multiplication by 1.

Reply:
there's also MAD :crossed_fingers:

Reply:
Use a loop for this was a tiny bit faster for one file I've been testing with, and with nvbench there was no real difference observed switching to a loop. I simplified a good bit of the encoding.